File: src/decoders/news-information.ts
import type { NewsCategory, SessionHandle, NewsInformation } from "~/models";
import { decodeNewsItem } from "./news-item";
import { decodeNewsQuestion } from "./news-question";

export const decodeNewsInformation = (information: any, session: SessionHandle, categories: NewsCategory[]): NewsInformation => {
  return {
    is: "information",
    ...decodeNewsItem(information, categories),
    question: decodeNewsQuestion(information.listeQuestions.V[0], session),

    get attachments () {
      return this.question.attachments;
    },

    get acknowledged () {
      return this.question.answered;
    },

    get acknowledgedDate () {
      return this.question.answerDate;
    },

    get needToAcknowledge () {
      return this.question.shouldAnswer;
    },

    get content () {
      return this.question.content;
    }
  };
};

File: library/src/commonMain/kotlin/decoders/news-information.kt
package decoders

import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.jsonArray
import kotlinx.serialization.json.jsonObject
import models.NewsInformation
import models.SessionInformation

fun decodeNewsInformation (information: JsonObject, sessionInformation: SessionInformation): NewsInformation {
    return NewsInformation(
        question = decodeNewsQuestion(information["listeQuestions"]!!.jsonObject["V"]!!.jsonArray[0].jsonObject, sessionInformation)
    )
}